### ISSUE TITLE
How is clear but Why is not

### ISSUE THREAD

**Issue by dee-cf** [dotnet/dotnet-docker-samples#72]
Could someone in plain english explain why would i, for example, choose to put my asp.net core application into Docker container vs putting it into Azure AppService?

I understand How's but don't have a clear pictures about Why's.

To be clear, i like the idea and have tried it already but for now it was just out of curiousity. Dont know what would be the main reasons for doing so?

thanks

**Comment by TheValkyrie** [comment]
Flexibility in hosting platforms, increased density potential, throughout the lifecycle all the bits in any environment are the same bits as in any other environment.
Mark West
[Image result for New Microsoft Logo][Image result for heart icon][https://tse1.mm.bing.net/th?&id=OIP.Mf7d3acfaf371a7e09dcc0e27784d445do0&w=132&h=131&c=0&pid=1.9&rs=0&p=0&r=0]

**Comment by kendrahavens** [comment]
I want to separate this question into two parts to avoid confusion:

1. Why use Docker containers?
2. Why use Docker containers in Azure App Service versus regular Azure App Service (that uses a vm)?

To answer, "Why use Docker containers?" I'd recommend this [blog post](https://blogs.msdn.microsoft.com/dotnet/2017/05/25/using-net-and-docker-together/) that highlights the benefits of containers. As mentioned above, they are lighter-weight (increased density potential) and more flexible than VMs. At the most basic level this translates into saving a lot of time and money on configuring and running servers. The concepts that Docker introduces can also save a lot of dev and ops time because it provides better organization and flexibility to deployment. The post also goes over different .NET scenarios that are useful to use with containers. Docker has a lot of [great material](https://www.docker.com/what-docker) covering the huge benefits of containers.

To answer, "Why use Docker containers in Azure App Service versus regular Azure App Service (that uses a vm)?" the container option *will have a cheaper cost versus running the same app on a virtual machine. 

*This is, however, obviously determined by the app and pricing model. Business decisions on pricing models aside, a dockerized app is generally cheaper to run anywhere. Though the pricing model is now different with the introduction of [Azure Container Instances (ACI)](https://azure.microsoft.com/en-us/blog/announcing-azure-container-instances/). I don't have a flat out comparison on pricing because it will depend on the needs of each individual app. Many customers and internal Microsoft teams have seen costs of building and running apps in production reduced by moving the app to containers.

**Comment by kendrahavens** [comment]
@codefactorydevelopment Thank you for asking this question. It's great to have clear answers to these scenarios in a space that is developing so rapidly. I'll close this issue if no further explanation is needed.

**Comment by dee-cf** [comment]
@kendrahavens

thank you for your answer! it definitely gives some clear reasons for different options.

i have no further questions at the moment.

thank you!
